src/main/java/org/reaktivity/nukleus/oauth/internal/stream/OAuthProxyFactory.java

```diff
@@ -678,8 +769,31 @@ private void onTokenExpiredSignal(
 
             if (delay > 0)
             {
-                this.expiryFuture = executor.schedule(delay, MILLISECONDS,
-                        targetRouteId, targetStreamId, TOKEN_EXPIRED_SIGNAL);
+                final long challengeAfter = grant.challengeAfter();
+                final boolean hasChallengeCapability = Capabilities.canChallenge(capabilities);
```

Comment:
This can be inlined to the `if` statement directly.
Use a static import for `Capabilities.canChallenge` so the `if` becomes `if (canChallenge(capabilities))`.